addons/base_import/models/base_import.py

```diff
@@ -453,10 +473,19 @@ def _match_headers(self, rows, fields, options):
             return [], {}
 
         headers = next(rows)
-        return headers, {
-            index: [field['name'] for field in self._match_header(header, fields, options)] or None
-            for index, header in enumerate(headers)
-        }
+        matches = {}
+        mapping_fields = self.env['base_import.mapping'].search_read([('res_model', '=', self.res_model)], ['column_name', 'field_name'])
+        for index, header in enumerate(headers):
+            match_fields = []
+            for mapping_field in mapping_fields:
+                if mapping_field['column_name'] == header.lower():
+                    match_fields = mapping_field['field_name'].split('/') if mapping_field['field_name'] else []
+            if not match_fields:
+                for field in self._match_header(header, fields, options):
+                    if field['name']:
+                        match_fields.append(field['name'])
```

Comment:
`_match_headers` seem to assume all fields will have a name, so this seems completely unnecessary, just use e.g.

    match_fields = [f['name'] for f in self._match_header(…)]